File: jushi_agent/src/backend/models/feishu.ts
/**
 * 飞书相关数据模型
 */

export interface OAuthCallbackRequest {
  code: string;
  state?: string;
  redirect_uri?: string;
}

export interface FeishuUserInfo {
  open_id: string;
  union_id: string;
  name: string;
  avatar_url?: string;
  email?: string;
  mobile?: string;
}

export interface FeishuTokenInfo {
  access_token: string;
  token_type: string;
  expires_in: number;
  scope?: string;
  refresh_token?: string;
  refresh_expires_in?: number;
}

export interface FeishuCalendar {
  calendar_id: string;
  summary: string;
  description?: string;
  permissions: string;
  color?: string;
  type: string;
  summary_alias?: string;
  is_deleted: boolean;
  is_third_party: boolean;
  role: string;
  is_primary: boolean;
}

export interface FeishuCalendarEvent {
  event_id: string;
  summary: string;
  description?: string;
  start_time?: {
    timestamp: string;
    timezone: string;
  };
  end_time?: {
    timestamp: string;
    timezone: string;
  };
  vchat?: any;
  visibility?: string;
  attendee_ability?: string;
  free_busy_status?: string;
  location?: {
    name: string;
    address?: string;
  };
  color?: number;
  reminders?: Array<{
    minutes: number;
  }>;
  recurrence?: any;
  status?: string;
  is_exception?: boolean;
  meeting_room?: any;
  app_link?: string;
  attendees?: Array<{
    open_id: string;
    response_status: string;
  }>;
  organizer?: {
    open_id: string;
  };
  need_notification?: boolean;
  create_time?: string;
  update_time?: string;
}

export interface ApiResponse<T = any> {
  success: boolean;
  message?: string;
  data?: T;
  error?: string;
  code?: string;
  timestamp?: string;
}

export interface FeishuCalendarListResponse {
  calendars: FeishuCalendar[];
  has_more: boolean;
  page_token?: string;
  sync_token?: string;
}

export interface FeishuCalendarEventsResponse {
  items: FeishuCalendarEvent[];
  has_more: boolean;
  page_token?: string;
  sync_token?: string;
}

File: jushi_backend/app/models/feishu.py
"""
飞书相关数据模型
"""

from pydantic import BaseModel
from typing import Optional, List, Dict, Any

class OAuthCallbackRequest(BaseModel):
    """OAuth回调请求模型"""
    code: str
    state: Optional[str] = None
    redirect_uri: Optional[str] = None

class FeishuUserInfo(BaseModel):
    """飞书用户信息模型"""
    open_id: str
    union_id: str
    name: str
    avatar_url: Optional[str] = None
    email: Optional[str] = None
    mobile: Optional[str] = None

class FeishuTokenInfo(BaseModel):
    """飞书令牌信息模型"""
    access_token: str
    token_type: str = "Bearer"
    expires_in: int
    scope: Optional[str] = None
    refresh_token: Optional[str] = None
    refresh_expires_in: Optional[int] = None

class FeishuCalendar(BaseModel):
    """飞书日历模型"""
    calendar_id: str
    summary: str
    description: Optional[str] = None
    permissions: str
    color: Optional[str] = None
    type: str
    summary_alias: Optional[str] = None
    is_deleted: bool = False
    is_third_party: bool = False
    role: str
    is_primary: bool = False

class FeishuCalendarEvent(BaseModel):
    """飞书日历事件模型"""
    event_id: str
    summary: str
    description: Optional[str] = None
    start_time: Optional[Dict[str, Any]] = None
    end_time: Optional[Dict[str, Any]] = None
    vchat: Optional[Dict[str, Any]] = None
    visibility: Optional[str] = None
    attendee_ability: Optional[str] = None
    free_busy_status: Optional[str] = None
    location: Optional[Dict[str, Any]] = None
    color: Optional[int] = None
    reminders: Optional[List[Dict[str, Any]]] = None
    recurrence: Optional[Dict[str, Any]] = None
    status: Optional[str] = None
    is_exception: Optional[bool] = None
    meeting_room: Optional[Dict[str, Any]] = None
    app_link: Optional[str] = None
    attendees: Optional[List[Dict[str, Any]]] = None
    organizer: Optional[Dict[str, Any]] = None
    need_notification: Optional[bool] = None
    create_time: Optional[str] = None
    update_time: Optional[str] = None

class ApiResponse(BaseModel):
    """通用API响应模型"""
    success: bool
    message: Optional[str] = None
    data: Optional[Any] = None
    error: Optional[str] = None
    code: Optional[int] = None
